trigger/project_triggers.go

```diff
@@ -104,9 +118,13 @@ func metadataFromVersion(args ProcessorArgs) (model.VersionMetadata, error) {
 		return metadata, errors.Errorf("repo '%s' not found", args.DownstreamProject.Id)
 	}
 	metadata.Revision.Revision = repo.LastRevision
-	author, err := user.FindOneById(args.SourceVersion.AuthorID)
+	var authorId string
+	if args.SourceVersion != nil {
+		authorId = args.SourceVersion.AuthorID
+	}
+	author, err := user.FindOneById(authorId)
```

Comment:
I think we should only call this in the if statement as well, or else it's a waste of a DB call; we could just set author as a var outside so we can still use the if author != nil check either way